test/compile_execute_file/bit_overflow.c.sh

```diff
@@ -0,0 +1,30 @@
+#!/bin/bash
```

Comment:
It is great that you have these scripts. Now what we need is a way to include them in the CI.

Please can you have a look at:

- `scripts/check_single_file_tests.sh`
- `check_all.sh`

I wonder if you might make something called:

- `scripts/check_compile_execute_file.sh`

This should run all of the `.sh` scripts of the form you have added (there are only two now, but we may add more later).

Then update the `check_all.sh` script so that it runs the new `check_compile_execute_file.sh` script.

That should ensure that this all runs in the CI.

Reply:
It's probably worth compiling the examples with expected output with `-fsanitize=undefined` as I believe this would have also caught this error and if no mutants are enabled, Dredd shouldn't introduce any undefined behavior.

Reply:
@JamesLee-Jones Good idea.

Reply:
@afd I saw you are implementing some kind of CI test at [PR 255](https://github.com/mc-imperial/dredd/pull/255). Does it overlap with `check_compile_execute_file.sh` mentioned here?

Reply:
Yes, I think it probably will. I should get #255 landed soon. Once it's done, perhaps you can rework your PR to use the same infrastructure that I am bringing in there.